tests/eas/cpuhp.py

```diff
@@ -0,0 +1,217 @@
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os.path
+import random
+import json
+import sys
+
+from test import LisaTest, experiment_test
+from target_script import TargetScript
+from devlib.module.hotplug import HotplugModule
+
+class _CpuHotplugTest(LisaTest):
+    """
+    "Abstract" base class for generic EAS tests under CPU hotplug stress
+
+    Subclasses should provide a .workloads member to populate the 'wloads' field
+    of the experiments_conf for the Executor. Furthermore, subclasses may
+    overwrite the .hp_stress member containing the configuration used while
+    generating the random hotplug stress sequence. 
+
+    The test will be successful if the system does not halt or kill the
+    workloads because of hotplug stress. It means that the test passes if the
+    workload completes.
+    """
+    
+    test_conf = {
+        "ftrace" : {
+            "events" : [
+                "sched_overutilized",
+                "sched_energy_diff",
+                "sched_load_avg_task",
+                "sched_load_avg_cpu",
+                "sched_migrate_task",
+                "sched_switch",
+                "cpu_frequency",
+                "cpu_idle",
+                "cpu_capacity",
+                "cpuhp_enter",
+                "cpuhp_exit",
+                "cpuhp_multi_enter",
+            ],
+        },
+        "modules": ["cgroups", "hotplug"],
+        "tools": ["rt-app", "trace-cmd"],
+    }
+
+    hp_stress = {
+        'seed' : None,                  # Seed of the random number generator
+        'sequence_len' : 100,           # Number of operations in the sequence
+        'sleep' : {
+            'min_ms' : 10,              # Min sleep duration between hotplugs 
+            'max_ms' : 100,             # Max sleep duration between hotplugs
+        },                              #   max_ms <= 0 will encode 'no sleep'
+        'max_cpus_off' : sys.maxint,    # Max number of CPUs plugged-off
+    }
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(_CpuHotplugTest, cls).runExperiments(*args, **kwargs)
+
+    @classmethod
+    def _getExperimentsConf(cls, test_env):
+        conf = {
+            'tag' : 'energy_aware',
+            'flags' : ['ftrace', 'freeze_userspace'],
+            'sched_features' : 'ENERGY_AWARE',
+        }
+
+        return {
+            'wloads' : cls.workloads,
+            'confs' : [conf],
+        }
+
+    @classmethod
+    def _experimentsInit(cls, *args, **kwargs):
+        # Choose a random seed explicitly if not given
+        if cls.hp_stress.get('seed') is None:
+            random.seed() 
+            cls.hp_stress['seed'] = random.randint(0, sys.maxint)
+        random.seed(cls.hp_stress['seed'])
+        cls.te._log.info('Random sequence of CPU Hotplug generated with: ')
```

Comment:
Should we better use a local logger, which explicitly reports which class we are.
Something like:
   
```python
# Setup logging
        self._log = logging.getLogger('CpuHotplug')
```

... and just add the new channel `CpuHotplug` to the logger configuration:
https://github.com/ARM-software/lisa/blob/master/logging.conf
by basically copying/updating everything has TestEnv in it ;-)

Reply:
Done